internal/provider/kubernetes/controller.go

```diff
@@ -129,13 +129,25 @@ func newGatewayAPIController(mgr manager.Manager, cfg *config.Server, su Updater
 	}
 	r.log.Info("created gatewayapi controller")
 
-	// Subscribe to status updates
-	r.subscribeAndUpdateStatus(ctx, cfg.EnvoyGateway.EnvoyGatewaySpec.ExtensionManager != nil)
-
 	// Watch resources
 	if err := r.watchResources(ctx, mgr, c); err != nil {
 		return fmt.Errorf("error watching resources: %w", err)
 	}
+
+	// When leader election is enabled, only subscribe to status updates upon acquiring leadership.
+	if cfg.EnvoyGateway.Provider.Type == egv1a1.ProviderTypeKubernetes &&
+		!ptr.Deref(cfg.EnvoyGateway.Provider.Kubernetes.LeaderElection.Disable, false) {
+		go func() {
+			select {
+			case <-ctx.Done():
+				return
+			case <-mgr.Elected():
```

Comment:
afaik this won't work, you can only receive a value from a channel once
you can use `mgr.Elected().Done(func() {
  r.subscribeAndUpdateStatus(ctx, cfg.EnvoyGateway.EnvoyGatewaySpec.ExtensionManager != nil)
})`

or we'll need to move this func to https://github.com/envoyproxy/gateway/blob/23856723c58df0ad28e128481f558bce41078700/internal/provider/kubernetes/kubernetes.go#L110 where the handle exists

Reply:
my preference would be to replace the existing svr.Elected with a waitGroup which is a more easier way to signal once based on `mgr.Elected()` and have every one else (infra mgr, watchable status) wait until then
cc @alexwo